### ISSUE TITLE
Case of 

### ISSUE THREAD

**Issue by KAction** [leahneukirchen/redo-c#21]
Here is short script that demonstrates some behavior that I find unexpected:

```
#!/bin/sh
tempdir=$(mktemp -d)
cd "${tempdir}"

echo "redo-ifchange v1" > v2.do
echo "redo-ifchange file" > v1.do

cat << 'EOF' > file.do
redo-ifchange file.src
cat file.src > $3
EOF

echo 1 > file.src
redo v2
cmp file.src file

echo 2 > file.src
redo v2
cmp file.src file
```

It created directory with build system, representing chain of dependencies "v2 -> v1 -> file". When `redo v2` is invoked first time, it correctly builds "file" from "file.src", but after "file.src" is changed, it fails to regenerate "file" and thinks that "v2" is up-to-date.

I am not sure whether I am misunderstanding how virtual rules are supposed to work or it is really bug.

**Comment by leahneukirchen** [comment]
Can't reproduce with 7eede3b:

```
rhea /tmp/fooo% echo "redo-ifchange v1" > v2.do
rhea /tmp/fooo% echo "redo-ifchange file" > v1.do
rhea /tmp/fooo% 
rhea /tmp/fooo% cat << 'EOF' > file.do
heredoc> redo-ifchange file.src
heredoc> cat file.src > $3
heredoc> EOF
rhea /tmp/fooo% echo 1 > file.src
rhea /tmp/fooo% redo v2
redo v2 # ./v2.do
redo   v1 # ./v1.do
redo     file # ./file.do
rhea /tmp/fooo% cmp file.src file
rhea /tmp/fooo% echo 2 > file.src
rhea /tmp/fooo% redo v2
redo v2 # ./v2.do
redo   v1 # ./v1.do
redo     file # ./file.do
rhea /tmp/fooo% cmp file.src file
rhea /tmp/fooo% 
```

(my prompt would have shown `1?` if a cmp failed)

**Comment by leahneukirchen** [comment]
Aha, with the script I can. Inserting "sleep 2" before changing file.src fixes it, so it must be an error in timestamp detection...

**Comment by leahneukirchen** [comment]
I think this was introduced in 6beed5c.

**Comment by leahneukirchen** [comment]
Please try the branch `msec`.

**Comment by KAction** [comment]
Yes, it passes with that branch, but I still have feeling that something deeper is wrong. Isn't rebuild supposed to happen due hash sum mismatch, no matter the timestamp?

**Comment by leahneukirchen** [comment]
The timestamp check was added as a short cut since recomputing all hashes all the time is quite expensive...

**Comment by KAction** [comment]
Didn't check code yet. Is it possible to set flag saying that I am okay to pay that price for more reliable rebuilds?

**Comment by leahneukirchen** [comment]
you can comment out the first line there :)

**Comment by KAction** [comment]
Well, I can, but probably we can make it configurable with not so big complexity cost. What about #23?
